### ISSUE TITLE
Call for Input: Merge EIP Describing Flaws in ERC-20

### ISSUE THREAD

**Issue by SamWilsn** [ethcatherders/EIPIP#293]
## Call for Input

<table>
<tr>
<th>Decision</th>
<td>

Do we merge https://github.com/ethereum/EIPs/pull/7915 ?

</td>
</tr>
<tr>
<th>If Affirmed</th>
<td>

A new Informational EIP is created describing apparent flaws in the ERC-20 standard.

</td>
</tr>
<tr>
<th>If Rejected</th>
<td>No new EIP is created.</td>
</tr>
<tr>
<th>Method</th>
<td>Rough Consensus</td>
</tr>
<tr>
<th>Deadline</th>
<td>December 9, 2023</td>
</tr>
</table>

## Background

Probably should read https://github.com/ethereum/EIPs/pull/7915 for the full details.

tl;dr `transfer` allows tokens to be sent to contracts that don't support tokens.

**Comment by SamWilsn** [comment]
I am **opposed** to merging this pull request.

---

Please see https://ethereum-magicians.org/t/modification-of-eip-process-to-account-for-security-treatments/16265/8 for my reasoning.

**Comment by abcoathup** [comment]
I am against merging (but I don't have a vote).

If this were to be merged it should be in the ERC repo as it is about an app layer.

Though I would much prefer a wiki page for each ERC.  
An ERC wiki page could have the latest information about security and implementations and could be updated and maintained by the community.

**Comment by Dexaran** [comment]
@abcoathup 

>Though I would much prefer a wiki page for each ERC.
An ERC wiki page could have the latest information about security and implementations, be updated/maintained by the community. This would remove the issue of updating final EIPs/ERCs.

This is not a correct way to provide sensitive security information **if we want to prioritize security of our users.** And I think we MUST prioritize security as we are dealing with financial software.

If some object is particularly dangerous then the warning label must be placed on this exact object.

In order to prevent people from getting bleach poisoning the danger warning is put on the bleach bottle. You can't create a separate page where you list all the dangerous liquids and foster discussions while pretending that every human will go and browse the page before drinking any liquid from any bottle. This is simply not how things work.

**Comment by Dexaran** [comment]
@SamWilsn's claims are irrelevant here.

>I am opposed to merging this pull request.
>Please see https://ethereum-magicians.org/t/modification-of-eip-process-to-account-for-security-treatments/16265/8 for my reasoning.


In the linked thread he said the following:

>"I object to including security considerations surfaced after a proposal becomes final within the body of the proposal itself. To be clear, I do believe that these security considerations should be published somewhere, just not within the EIP.
My objection stems from the question of who has the authority to determine if a security disclosure is worthy of publishing.


This PR https://github.com/ethereum/EIPs/pull/7915 does not propose any changes to any EIP. It only creates a new informational EIP as it was suggested by the EIP editors before.

### First, I would blame EIP editors for judging my proposal in the DRAFT status. 

As I can read in the [EIP-1](https://eips.ethereum.org/EIPS/eip-1#eip-editors) and [EIP-5069](https://eips.ethereum.org/EIPS/eip-5069) you should:

- Read the EIP to check if it is ready: sound and complete. The ideas must make technical sense, even if they don’t seem likely to get to final status.
- The title should accurately describe the content.
- Check the EIP for language (spelling, grammar, sentence structure, etc.), markup (GitHub flavored Markdown), code style

And this is what you should not:

![EIP5069](https://github.com/ethereum-cat-herders/EIPIP/assets/26142412/7dfdfb71-6d4c-4cb6-8e83-51de32375c53)

I don't see any mentions of any reasons that would prevent me from creating a proposal that does make technical sense. I think that it can be accepted as DRAFT as soon as the language is correct. I would be fine with EIP editors merging it as DRAFT and expressing their concerns regarding the correctness of the proposal during the PEER REVIEW stage. I would be fine if the proposal never gets finalized until all the expressed concerns are resolved.

However I would blame EIP editors for **"deciding the winner"** here. By not allowing me to publish sensitive info related to one of the proposals you are implying that the proposal is good - which in fact is not true because it caused [$90,000,000 to $200,000,000 millions of dollars financial damage](https://dexaran.github.io/erc20-losses) to our users.

Also, if you don't have a process or a security expert in your team  then it would be much better to let all the security-related proposals to get merged as DRAFTs and never let them past REVIEW in long term. Because if a security expert will once join your team and you will assign him to security-related EIPs and ERCs they will all be there in the DRAFT statuses.

### Second, I encourage EIP editors to think about the mission here

We are not discussing random proposal. **We are discussing a real problem.** A problem that damages people, endangers their wellbeing and may be even ruins lives.

- This is how this people look and this is what they write: https://www.reddit.com/r/0xProject/comments/7lpc5n/ive_sent_my_tokens_to_the_zrx_smart_contract/
- This is how this people look: https://twitter.com/Crypto_Plato/status/1303235432755294216
- This is how this people look: https://ethereum.stackexchange.com/questions/34559/i-accidentally-sent-a-token-to-a-token-contract-can-i-get-it-back

This problem can be solved technically. This has persisted for years because of everyone's reluctance to take responsibility for announcing it.

Imagine that you are a bank and you are using a database that contains a flaw that was discovered 6 years ago but nobody took responsibility of upgrading yours so you are still using it. As the result your customers lost $90,000,000 to software bugs but you decided to hide this fact. Does it look fair? Would you put your money in such bank? What will people think of it once it gets discovered?

![EX26](https://github.com/ethereum-cat-herders/EIPIP/assets/26142412/6e98277f-ce3e-4cae-b522-2bee91d62534)

How do you "foster transparency and ensure that valuable insights from past proposals are accessible" if you silence a disclosure of a catastrophic problem that caused real financial damage that exceeds the scale of TheDAO hack?

You can probably call multiple reasons for simply ignoring the problem like "we EIP editors don't want to decide XYZ" but you know the problem is still there. It will not just disappear on its own, it will be there in 2024 if we don't make some steps to solve it now.

**Comment by xinbenlv** [comment]
I am *opposing* to merging this pull request.

---------

As I have posted on [this comment](https://github.com/ethereum/EIPs/pull/7915#issuecomment-1805072404 ) 
I don't think a new Information ERC is helpful in presenting the warning. I offered my recommendation of alternatives in my comment for author to pursue.

**Comment by Joeysantoro** [comment]
I am opposed to merging this pull request at this time:
1. There is an ongoing discussion on how security considerations should fold into the EIP process that is far from resolved: https://ethereum-magicians.org/t/modification-of-eip-process-to-account-for-security-treatments/16265
2. This particular proposal is too opinionated on the description and resolution of the problem. Many of the solutions proposed in the Security Considerations section are unworkable, force the modification of a Final EIP in nontrivial ways, or introduce other security/centralization concerns.

**Comment by SamWilsn** [comment]
This is well past its deadline, so last call for comments from editors: @lightclient, @axic, @gcolvin, @g11tech.

**Comment by g11tech** [comment]
Opposed, a new EIP proposing improvements can be done

**Comment by SamWilsn** [comment]
While I'm not happy closing this issue without input from more Editors, the prevailing opinion in this thread is to not merge this pull request.
